embeddings/pipeline/hps_pipeline.py

```diff
@@ -0,0 +1,341 @@
+    def _get_evaluation_metadata(self, parameters: SampledParameters) -> EvaluationMetadata:
+        pass
+
+    def get_best_paramaters(self, study: Study) -> Metadata:
+        best_params = study.best_params
+        parsed_params = self.config_space.parse_parameters(best_params)
+        return self._get_metadata(parsed_params)
+
+    def run(
+        self,
+    ) -> Tuple[pd.DataFrame, Metadata]:
+        self._pre_run_hook()
+        self.preprocessing_pipeline.run()
+        study: Study = optuna.create_study(
+            direction="maximize",
+            sampler=self.sampler,
+            pruner=self.pruner,
+        )
+        study.optimize(self.objective, n_trials=self.n_trials, show_progress_bar=True)
+
+        if isinstance(self.dataset_path, TemporaryDirectory):
+            self.dataset_path.cleanup()
+
+        metadata: Metadata = self.get_best_paramaters(study)
+        self._post_run_hook()
+        return study.trials_dataframe(), metadata
+
+    def objective(self, trial: optuna.trial.Trial) -> float:
+        parameters = self.config_space.sample_parameters(trial=trial)
+        parsed_params = self.config_space.parse_parameters(parameters)
+        args = self._get_evaluation_metadata(parsed_params)
+
+        pipeline = self.evaluation_pipeline(**args)
+        results = pipeline.run()
+        metric = results[self.metric_name][self.metric_key]
+        assert isinstance(metric, float)
+        return metric
+
+    def _pre_run_hook(self) -> None:
+        logging.getLogger("flair").setLevel(logging.WARNING)
+
+    def _post_run_hook(self) -> None:
+        logging.getLogger("flair").setLevel(logging.INFO)
+
+
+@dataclass  # type: ignore
+class BaseHuggingFaceOptimizedPipeline(ABC):
+    dataset_name: str
+    embedding_name: str
+    input_column_name: str
+    target_column_name: str
+    n_warmup_steps: int = 10
+    n_trials: int = 2
+    sample_dev_split_fraction: Optional[float] = 0.1
+    seed: int = 441
+    fine_tune_embeddings: bool = False
+    pruner_cls: Type[optuna.pruners.MedianPruner] = field(
+        init=False, default=optuna.pruners.MedianPruner
+    )
+    sampler_cls: Type[optuna.samplers.TPESampler] = field(
+        init=False, default=optuna.samplers.TPESampler
+    )
+
+    @abc.abstractmethod
+    def __post_init__(self) -> None:
+        pass
+
+
+@dataclass
+class OptimizedFlairClassificationPipeline(
+    OptunaPipeline[
+        FlairModelTrainerConfigSpace,
+        HuggingFaceClassificationPipelineMetadata,
+        EvaluationPipelineMetadata,
+    ],
+    BaseHuggingFaceOptimizedPipeline,
+):
+    dataset_dir: TemporaryDirectory[str] = field(init=False, default=TemporaryDirectory())
+    tmp_model_output_dir: TemporaryDirectory[str] = field(init=False, default=TemporaryDirectory())
+    config_space: FlairModelTrainerConfigSpace = FlairModelTrainerConfigSpace()
+
+    def __post_init__(self) -> None:
+        self.dataset_path = Path(self.dataset_dir.name).joinpath("ds.pkl")
```

Comment:
It's because FlairTextClassification corpus can be mapped to single file, in this case with pickle persister. And the sequence labeling need directory.